pkg/apis/kops/cluster.go

```diff
@@ -486,3 +487,17 @@ func (z *ClusterZoneSpec) assignCIDR(c *Cluster) (string, error) {
 func (c *Cluster) SharedVPC() bool {
 	return c.Spec.NetworkID != ""
 }
+
+// --------------------------------------------------------------------------------------------
+// Network Topology functions for template parsing
+//
+// Each of these functions can be used in the model templates
+// The go template package currently only supports boolean
```

Comment:
You can do `eq Topology "private"` I think.

In general, it might be easier to replace the network.yaml with go code.  I have a prototype of it somewhere that I'll paste a link to if I find it.  But the whole reliance on templates just seems pretty fragile and seems to have turned out to be harder for people to change, not easier.

But completely up to you if you want to do this for this PR.  Just if you're fighting templates you might find it easier to dump them :-)